src/java.base/share/classes/java/lang/invoke/GenerateJLIClassesHelper.java

```diff
@@ -53,16 +55,28 @@
     private static final String SPECIES_RESOLVE = "[SPECIES_RESOLVE]";
 
     static void traceLambdaForm(String name, MethodType type, Class<?> holder, MemberName resolvedMember) {
-        if (TRACE_RESOLVE) {
-            System.out.println(LF_RESOLVE + " " + holder.getName() + " " + name + " " +
+        if (TRACE_RESOLVE || VM.isDumpLoadedClassListSetAndOpen) {
+            String traceLF = LF_RESOLVE + " " + holder.getName() + " " + name + " " +
                     shortenSignature(basicTypeSignature(type)) +
-                    (resolvedMember != null ? " (success)" : " (fail)"));
+                    (resolvedMember != null ? " (success)" : " (fail)");
+            if (TRACE_RESOLVE) {
+                System.out.println(traceLF);
+            }
+            if (VM.isDumpLoadedClassListSetAndOpen) {
+                VM.cdsTraceResolve(traceLF);
+            }
```

Comment:
GenerateJLIClassesHelper shouldn't need to know why the trace is needed. Also, "cdsTraceResolve" is too generic.

I think it's better to have
```
if (TRACE_RESOLVE || VM.CDS_TRACE_JLINV_RESOLVE) {
    ...
    VM.cdsTraceJLINVResolve(traceLF);
```

The acronym JLINV is used in [methodHandles.cpp](https://github.com/openjdk/jdk/blob/ce93cbce77e1f4baa52676826c8ae27d474360b6/src/hotspot/share/prims/methodHandles.cpp#L1524)

Reply:
With CDS related code moved to CDS.java, I think we should keep TRACE_RESOLVE here. A new name like suggested by Mandy, logTraceResolve in CDS.java